plugins/auth-backend-module-microsoft-provider/src/strategy.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassportProfile } from '@backstage/plugin-auth-node';
+import { decodeJwt } from 'jose';
+import fetch from 'node-fetch';
+import { Strategy as MicrosoftStrategy } from 'passport-microsoft';
+
+export class ExtendedMicrosoftStrategy extends MicrosoftStrategy {
+  userProfile(
+    accessToken: string,
+    done: (err?: Error | null, profile?: PassportProfile) => void,
+  ): void {
+    if (this.skipUserProfile(accessToken)) {
```

Comment:
minor nit: if the jwt decode fails or for other reasons an exception is thrown, `done` won't be called at all - could have a try/catch just to be sure. Same inside the profile callback below but that one seems to be protected already and should (TM) be un-crashable

Reply:
I can check that case. The issue might be what to use as a fallback. If the skipUserProfile call fails and we use false as default, the whole auth will still fail due to the foreign scope which it tries to prevent. Using true as default seems wrong, though.

Reply:
I've added a try-catch inside of the function with fallback to false which may then fail at fetching the profile in case of foreign scopes. Or due to the same issue which causes that one to fail.

The main reason why it could fail is an invalid JWT token. Not sure how likely that is at this point, though.